### PR TITLE
21 GUI based service conditions manipulation

### PR DESCRIPTION
**Pull request by baha-a**
~Depends on #24.
This PR **shouldn't** be merged to main before #24.~

### CODE REVIEW

**Review by 1AhmedYasser — APPROVED** [review]

File: GUI/src/RootComponent.tsx [hunk 1]
```diff
@@ -5,28 +5,23 @@ import NotFoundPage from './pages/NotFoundPage'
...
+import './styles/main.scss'
 
 const RootComponent: React.FC = () => {
   return (
     <Routes>
-      <Route path={ROUTES.NEWSERVICE_ROUTE} element={<NewServicePage />} />
-      <Route path={ROUTES.FLOW_ROUTE} element={<ServiceFlowPage />} />
       <Route element={<Layout />}>
+        <Route path={ROUTES.NEWSERVICE_ROUTE} element={<NewServicePage />} />
+        <Route path={ROUTES.FLOW_ROUTE} element={<ServiceFlowPage />} />
```
Comment (Minwasko) [inline GUI/src/RootComponent.tsx:18]: Please move these 2 routes outside of layout element as they previously were. 
In the design these pages don't have a menu

File: GUI/src/components/APIEndpointCard/index.tsx [hunk 2]
```diff
@@ -14,7 +14,7 @@ import {
 import { Option } from "../../types/option";
 import { useTranslation } from "react-i18next";
 import { MdDeleteOutline } from "react-icons/md";
-import "./ApiEndpointCard.scss";
+import "./APIEndpointCard.scss"
```
Comment (Minwasko) [inline GUI/src/components/APIEndpointCard/index.tsx:17]: Please revert this change back to `import "./ApiEndpointCard.scss";` and rename the directory to `ApiEndpointCard`

File: GUI/src/components/FlowElementsPopup/ConditionInput.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  }, [rule.name])
+
+
+  const [_, drop] = useDrop(
+    () => ({
+      accept: 'tags',
+      drop(_item: any, monitor) {
+        const didDrop = monitor.didDrop()
+        if (didDrop) return;
+        setName("name + ' ' + _item.value")
```
Comment (Minwasko) [inline GUI/src/components/FlowElementsPopup/ConditionInput.tsx:25]: Is it supposed to be like that? Currently it's just a string

File: GUI/src/components/index.tsx [hunk 4]
```diff
@@ -30,11 +26,16 @@ import MainNavigation from "./MainNavigation";
...
 import Section from "./Section";
 import Toast from "./Toast";
 import Tooltip from "./Tooltip";
 import Track from "./Track";
+import FlowElementsPopup from "./FlowElementsPopup";
+import ApiEndpointCard from "./APIEndpointCard";
+import EndpointOpenAPI from "./APIEndpointCard/Endpoints/OpenAPI";
+import EndpointCustom from "./APIEndpointCard/Endpoints/Custom";
+import EndpointResqlComponent from "./APIEndpointCard/Endpoints/ResqlComponent";
+import RequestVariables from "./APIEndpointCard/Endpoints/RequestVariables";
```
Comment (Minwasko) [inline GUI/src/components/index.tsx:38]: Please revert changes to `ApiEndpointCard` and elements in it's directory. Let's make sure that they are all under `/ApiEndpointCard`

File: GUI/src/components/FlowElementsPopup/index.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+              <RulesBuilder
+                rules={rules}
+                setRules={setRules}
+              />
+            }
+          </Track>
+        }
+      </DndProvider>
+
+      {type !== 'input' &&
```
Comment (Minwasko) [inline GUI/src/components/FlowElementsPopup/index.tsx:87]: You can remove this part, it was just for testing and to show that it works.

File: GUI/src/pages/ServiceFlowPage.tsx [hunk 6]
```diff
@@ -88,30 +98,22 @@ const ServiceFlowPage: FC = () => {
...
-      )}
+      <FlowElementsPopup
+        onClose={() => setVisiblePopupNode(null)}
+        onSave={(rules: any) => {
+          setUpdatedRules(rules)
+          setVisiblePopupNode(null)
+        }}
+        node={visiblePopupNode}
+        oldRules={updatedRules}
+        addRuleCount={() => setUpdatedRules([null, null, null])}
```
Comment (Minwasko) [inline GUI/src/pages/ServiceFlowPage.tsx:115]: You can remove addRuleCount logic if it's not needed. It was there just for testing and to show how that it works

**Review by Minwasko — CHANGES_REQUESTED** [review]

**Review by Minwasko — APPROVED** [review]

**Review by trevorling — APPROVED** [review]

**Pull request merged**